packages/wrangler/src/pages/functions/buildWorker.ts

```diff
@@ -234,6 +239,58 @@ export function buildRawWorker({
 	);
 }
 
+export async function traverseAndBuildWorkerJSDirectory({
+	workerJSDirectory,
+	buildOutputDirectory,
+	d1Databases,
+	nodejsCompat,
+}: {
+	workerJSDirectory: string;
+	buildOutputDirectory: string;
+	d1Databases?: string[];
+	nodejsCompat?: boolean;
+}): Promise<BundleResult> {
+	const entrypoint = resolve(join(workerJSDirectory, "index.js"));
+
+	const traverseModuleGraphResult = await traverseModuleGraph(
+		{
+			file: entrypoint,
+			directory: resolve(workerJSDirectory),
+			format: "modules",
+			moduleRoot: resolve(workerJSDirectory),
+		},
+		[
+			{
+				type: "ESModule",
+				globs: ["**/*.js"],
+			},
+		]
+	);
+
+	const outfile = join(tmpdir(), `./bundledWorker-${Math.random()}.mjs`);
+	const bundleResult = await buildRawWorker({
+		workerScriptPath: entrypoint,
+		bundle: false,
+		outfile,
+		directory: buildOutputDirectory,
+		local: false,
+		sourcemap: true,
+		watch: false,
+		onEnd: () => {},
+		betaD1Shims: d1Databases,
+		nodejsCompat,
+	});
+
+	return {
+		modules: traverseModuleGraphResult?.modules ?? bundleResult?.modules ?? [],
```

Comment:
Under what case would it use `bundleResult?.modules`? Could you add a comment explaining what is happening here? I'm not as familiar with esbuild/etc and the way we're doing the traversal and then bundling (but not bundling) is confusing

Reply:
Yeah, good shout. This was me copy-pasting from elsewhere, but you're right. It'll never fallback to `bundleResult`s.